checkpointing.py

```diff
@@ -23,7 +28,17 @@ def get_checkpoint(self, collection):
             with open(CHECKPOINT_PATH) as checkpoint_store:
                 try:
                     checkpoint_list = json.load(checkpoint_store)
-                    start_time = checkpoint_list.get(collection)
+
+                    if not checkpoint_list.get(collection):
+                        self.logger.info(
+                            "The checkpoint file is present but it does not contain the start_time for the collection %s, hence considering the start_time and end_time from the configuration file instead of the last successful fetch time"
+                            % (collection)
+                        )
+                        start_time = self.data.get("start_time")
+                        end_time = self.data.get("end_time")
```

Comment:
Both the `start_time` and `end_time` are added it to provide more control to the user.
These are optional fields. So, if `start_time` is not provided, the code takes the default value (6 months from the current time) as `start_time`. If `end_time` is not provided, the code takes current time of code execution as `end_time` 

I got your point that the most common case (index everything), should happen by default if it is first index job. 
So, I'll see if i can make changes such that indexing happens from the beginning if `start_time` is not provided, instead of taking a fixed default value for `start_time`

 Thus, if the user chose to exclude both the start_time and end_time, everything is indexed from beginning to end in the first job. Thanks, for pointing it out.

Reply:
Update: we will need start_time as multiprocessing is implemented based on datetime partitions.